controllers/operands/networkAddons_test.go

```diff
@@ -558,29 +560,76 @@ var _ = Describe("CNA Operand", func() {
 
 			if o.ksdDeployExpected {
 				Expect(foundCNAO.Spec.KubeSecondaryDNS).ToNot(BeNil(), "KSD spec should be added")
+				Expect(foundCNAO.Spec.KubeSecondaryDNS.Domain).To(Equal(o.expectedBaseDomain),
+					"Expected domain should be set on KSD spec")
 			} else {
 				Expect(foundCNAO.Spec.KubeSecondaryDNS).To(BeNil(), "KSD spec should not be added")
 			}
-		},
-			Entry("should have KSD if feature gate is set to true", ksdAnnotationParams{
-				ksdExists:         false,
-				setFeatureGate:    true,
-				featureGateValue:  true,
-				ksdDeployExpected: true,
-			}),
-			Entry("should not have KSD if feature gate is set to false", ksdAnnotationParams{
-				ksdExists:         true,
-				setFeatureGate:    true,
-				featureGateValue:  false,
-				ksdDeployExpected: false,
-			}),
-			Entry("should not have KSD if feature gate does not exist", ksdAnnotationParams{
-				ksdExists:         true,
-				setFeatureGate:    false,
-				featureGateValue:  false,
-				ksdDeployExpected: false,
-			}),
-		)
+		}
+
+		Context("With K8s", func() {
+			DescribeTable("when reconciling kube-secondary-dns", ksdTester,
+				Entry("should have KSD if feature gate is set to true", ksdAnnotationParams{
+					ksdExists:          false,
+					setFeatureGate:     true,
+					featureGateValue:   true,
+					ksdDeployExpected:  true,
+					expectedBaseDomain: "",
+				}),
+				Entry("should not have KSD if feature gate is set to false", ksdAnnotationParams{
+					ksdExists:          true,
+					setFeatureGate:     true,
+					featureGateValue:   false,
+					ksdDeployExpected:  false,
+					expectedBaseDomain: "",
+				}),
+				Entry("should not have KSD if feature gate does not exist", ksdAnnotationParams{
+					ksdExists:          true,
+					setFeatureGate:     false,
+					featureGateValue:   false,
+					ksdDeployExpected:  false,
+					expectedBaseDomain: "",
+				}),
+			)
+		})
+
+		Context("With Openshift Mock", func() {
+			BeforeEach(func() {
+				getClusterInfo := hcoutil.GetClusterInfo
+
+				hcoutil.GetClusterInfo = func() hcoutil.ClusterInfo {
+					return &commonTestUtils.ClusterInfoMock{}
+				}
+
+				DeferCleanup(func() {
+					hcoutil.GetClusterInfo = getClusterInfo
+				})
+			})
```

Comment:
Possible idea:
Have the original GetClusterInfo always saved in a parameter of the core code,
and a function that will restore it
This way we can just call the function without creating a function, nor saving it

If there can be even a BeforeEach that we can call that does it all with the Defer it is even better
Can be adapted all over the code